### PR TITLE
DAOS-16874 tests: Update expected pool_layout_ver

### PR DESCRIPTION
**Pull request by liw**
Update expected pool_layout_ver in
dmg_pool_query_test/test_pool_query_basic.

Test-tag: test_pool_query_basic

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/control/dmg_pool_query_test.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def test_pool_query_basic(self):
                     "total": self.params.get("total", path="/run/exp_vals/nvme/*")
                 }
             ],
-            "pool_layout_ver": 3,
+            "pool_layout_ver": 4,
             "query_mask": self.params.get("query_mask", path="/run/exp_vals/*"),
             "upgrade_layout_ver": 3,
```
Comment (phender) [inline src/tests/ftest/control/dmg_pool_query_test.py:90]: Don't we also need to update the expected `upgrade_layout_ver`:
```suggestion
            "upgrade_layout_ver": 4,
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/control/dmg_pool_query_test.py (hunk 1 above)
Comment (liw) [reply src/tests/ftest/control/dmg_pool_query_test.py]: I think you're right. Thanks, Phil.

**Comment by liw** [comment]
Modified the commit message of the merge commit to add `test-tag`.

**Review by phender — APPROVED** [review]

**Comment by phender** [comment]
Testing looks good: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15610/3/testReport/FTEST_control/DmgPoolQueryTest/. It will require a forced landing due to only needing to run the one functional test.

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**